docs/developer.md

```diff
@@ -0,0 +1,312 @@
+:::
+
+To do local development, [clone the repository](https://github.com/jupyter-book/mystmd):
+
+```shell
+git clone git@github.com:jupyter-book/mystmd.git
+cd mystmd
+```
+
+Then, install dependencies via npm.
+You need to do this each time you pull from upstream, since dependency versions may change:
+
+```shell
+npm install
+```
+
+Now, build `mystmd`:
+
+```shell
+npm run build
+```
+
+Optionally, you can link the built executable as your globally installed `mystmd`:
+
+```shell
+npm run link
+```
+
+```{warning}
+The build process uses unix commands that might not work properly on Windows.
+When building on Windows, use either WSL or a unix-like shell (such as Git Bash or MSYS2), and make sure that npm is set to use these by default (`npm config set script-shell path/to/shell.exe`).
+```
+
+These commands allow you to use the `myst` CLI from any directory; source code changes are picked up after each `npm run build` (executed in the top-level source directory).
+
+### Developer workflow: myst-theme
+
+the [`myst-theme` README](https://github.com/jupyter-book/myst-theme/) provides a more detailed overview of the components of that package.
+
+Recall from the [architecture overview](#architecture) that `myst-md` is a React web application. It provides theming, and requires a separate content server for data. When developing, the steps are therefore to:
+
+1. Launch a content server
+2. Launch the `myst-theme` web application server (this is what you browse to)
+3. Run a process to monitor changes and rebuild `myst-theme`
+
+#### Content server
+
+We need some example data to test our theme against, such as [the example landing page](https://github.com/myst-examples/landing-pages) or ... Clone our data repository and start the content server:
+
+```shell
+git clone https://github.com/myst-examples/landing-pages
+cd landing-pages
+myst start --headless
+```
+
+(The `--headless` flag tells `myst` not to start up the theme server; we want to do that ourselves in the next step.)
+
+```{error}
+Can you help me with other good content site examples?
+```
+
+#### myst-theme server
+
+We now fire up the `myst-theme` React app, which will "hydrate" itself from the content server.
+
+First, clone the repository and install dependencies:
+
+```shell
+git clone https://github.com/jupyter-book/myst-theme/
+cd myst-theme
+npm install
+```
+
+Then, see if you can build the package:
+
+```shell
+npm run build
+```
+
+Good. Now let's launch the theme server:
+
+```shell
+npm run theme:book
+```
+
+After a while, it should tell you that the server has been started at http://localhost:3000. Browse there, and confirm that you can see the landing-page content.
+
+Each time after making changes to the myst-theme source, you'll need to recompile. You can do that using `npm run build`, but it may be easier to watch for changes and do that automatically:
+
+```shell
+npm run dev
```

Comment:
Could add a note here that you can also watch packages individually by changing directories and running that same command.